class/practice2.1/AdvancedHelloWorld.cpp

```diff
@@ -1,110 +1,94 @@
+    void LoadData() {
+        std::ifstream file(database);
+        std::string name;
+        int count;
+        while (file >> name >> count) {
+            statsData [name] = count;
+        }
+        file.close();
     }
-    if (WordToFind=="bread")
-    {
-        std::ofstream file("database.txt", std::ios::trunc); 
+    void SaveData() {
+        std::ofstream file(database, std::ios::trunc);
+        for (const auto& [name, count] : statsData ) {
+            file << name << " " << count << std::endl;
+        }
         file.close();
-        std::cout << "all data was cleared" << std::endl;
-        return 0;
     }
-    std::ifstream file("database.txt");
-    std::string line;
-    int lineNumber = 0;
-    bool found = false;
-    std::string lineFound;
-
-    std::vector<std::string> lines;
- 
-    while (std::getline(file, line)) {
-        lines.push_back(line);
+ std::string deleteUser(const std::string& nameToFind) {
+    auto name = statsData.find(nameToFind);
+    if (name != statsData.end()) {
+        statsData.erase(name);
     }
-    file.close();
-    for (const std::string& line : lines) {
-        lineNumber++;
-        if (line != "") {
-            if (splitString(line, ' ')[0] == WordToFind)
-            {
-                found = true;
-                lineFound = line;
-                break;
-            }
+    SaveData();
+    return "All data about this user was cleared";
+}
+
+    std::string UserIncrement(const std::string& nameToFind) {
+        std::string returnValue;
+        if (statsData.find(nameToFind)!=statsData.end())
+        {
+            statsData [nameToFind]++;
+            returnValue="Hello again(x" + std::to_string(statsData[nameToFind]) + ")," + nameToFind;
+        }else{
+            statsData [nameToFind]++;
+            returnValue="Welcome " +nameToFind;
         }
+        SaveData();
+        return returnValue;
     }
-    if (!found) {
-        if (argument2 == "delete") {
-            std::cout << "There is no such user " <<std::endl;
-            return 0;
-        }
-        std::cout << "Welcome " << WordToFind << std::endl;
-        std::ofstream fileEdit("database.txt", std::ios::app);
-        if (!fileEdit.is_open()) {
-            std::cerr << "Could not open the file for writing!" << std::endl;
-            return 1;
-        }
-        fileEdit << WordToFind << " 1" << std::endl;
-        fileEdit.close();
+public:
+    Database(const std::string& filePath) : database(filePath) {
+         LoadData();
     }
-    else {
-        if (argument2 == "delete") {
-
-            for (int i = 0; i < lines.size(); i++)
-            {
-                if (splitString(lines[i], ' ')[0] == WordToFind)
-                {
-                    lines.erase(lines.begin() + i);
-                }
+    void clearDatabase() {
+        statsData.clear();
+        SaveData();
+    }
+    std::string PerformChanges(const std::string& nameToFind, const std::string& additionalCommand) {
```

Comment:
it should be a part of business logic not a database. therefore database provides api to increment counter, delete user, clear database and so on. while business logic will parse commands and handle them to provide the expected behavior